src/main/groovy/lang/ObjectRange.java

```diff
@@ -537,8 +539,9 @@ protected Object increment(Object value) {
      * @param value the value to decrement
      * @return the decremented value
      */
-    protected Object decrement(Object value) {
-        return InvokerHelper.invokeMethod(value, "previous", null);
+    @SuppressWarnings("unchecked")
+    private T decrement(T value) {
```

Comment:
@paplorinc you did this on request, yes, but the pullrequest got a bit messy and it is really difficult to see all the implications before the actual change right now. Lot's of these problems stem also from the API here being such a mess.

There is two possibly problematic parts in this. One is the change from protected to private, the other is the change from Object to T. Since T has Comparable as base, we have a type change as well.  

Which actually leads me back to this API being a mess. Range is based on Comparable and this change was done in 2011... IntRange got changed to conform to this, but ObjectRange did not. 

And then let us look for example at get(int). The method is defined through AbstractList for once, Range sets the List type to "T extends Comparable", which means get(int) is supposed to return at least a Comparable. Now ObjectRange implements Range and defines get(int) as returning Object. This works only because Range is an interface. Had it been an abstract class, it would most probably define get(int) as returning Comparable and then ObjectRange would no longer be able to define an Object returning version. Fun fact:
`interface R<T extends Comparable> extends List<T>{}
class O extends AbstractList implements R{
  def get(int  i){null}
}` will in Groovy fail compilation with `The return type of java.lang.Object get(int) in O is incompatible with java.lang.Comparable in java.util.List`. And does rightfully do so. I really wonder why javac is not complaining here